client/src/components/DataFiles/DataFilesModals/DataFilesUploadModal.js

```diff
@@ -3,17 +3,26 @@ import { useSelector, useDispatch } from 'react-redux';
 import { useHistory, useLocation } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import { v4 as uuidv4 } from 'uuid';
-import {
-  Button,
-  Modal,
-  ModalHeader,
-  ModalBody,
-  ModalFooter,
-  Table
-} from 'reactstrap';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
 import { LoadingSpinner, FileInputDropZone } from '_common';
 import { findSystemOrProjectDisplayName } from 'utils/systems';
-import { FileLengthCell } from '../DataFilesListing/DataFilesListingCells';
+import DataFilesUploadModalListingTable from './DataFilesUploadModalListing/DataFilesUploadModalListingTable';
+
+import './DataFilesUploadModal.module.scss';
+
+export const DIRECTION_CLASS_MAP = {
+  vertical: 'is-vert',
+  horizontal: 'is-horz'
```

Comment:
What is the `horizontal` class supposed to look like? I'm seeing the same as `vertical`, `compact`
![Screen Shot 2021-02-04 at 5 57 28 PM](https://user-images.githubusercontent.com/20326896/106970706-13d99c00-6713-11eb-84b9-95bcba707c2e.png)

Reply:
This is the requirement:

Full size shows FileDropZone at the top and listing beneath it all the times

![](https://files.slack.com/files-pri/T03EBR1EB-F01JF1B3T8A/screen_shot_2021-01-11_at_10.23.42_am.png)

For the other two layouts: FileDropZone is showing in original size when listing is empty, when there’s content listing is showing at the top and FileDropZone is showing collapsed


![](https://files.slack.com/files-pri/T03EBR1EB-F01K4S4BXU0/screen_shot_2021-01-11_at_10.22.15_am.png)

![](https://files.slack.com/files-pri/T03EBR1EB-F01JF1CC3GS/screen_shot_2021-01-11_at_10.22.49_am.png)


I used the DescriptionList component as a guide, making similar implementation (two props: density and direction)  for consistency.

Reply:
The `<DescriptionList>` is a complex component. _I wrote that component, and even I think it is **too** complex 😄 ._

The code you have written, with `<DescriptionList>` as a guide, is difficult to understand. I will review, because I am familiar with `<DescriptionList>` **and** because I wrote the requirements.

Reply:
> Full size shows FileDropZone at the top and listing beneath it all the times

I see this one 👍 

> For the other two layouts: FileDropZone is showing in original size when listing is empty, when there’s content listing is showing at the top and FileDropZone is showing collapsed

I don't see this one being used anywhere.  Is it?  That would be a case where you used `FileInputDropZone` with the `file` being used (https://github.com/TACC/Core-Portal/blob/main/client/src/components/_common/Form/FileInputDropZone.js#L26), right?

* If it isn't being used somewhere in the code currently, is it planned to be used someplace 
that doesn't exist yet?  [If that is the case, maybe add tests to test it or add to `UI Patterns` area] 
* Or is this second layout is this referring to the uploading attachments for tickets [this is just me guessing :).  If so, i'm not certain what work needs to be done in this PR [note: as mentioned above FileInputDropZone with `files` seems to cover this, i think. 🤔 ]

Reply:
@nathanfranklin, current configuration for DataFilesUploadModal is `<DataFilesUploadModal direction="vertical" density="default" />`, that's why only one layout is shown. You can define this in `DataFilesModals`.
We have a test for that.
Let's wait on @tacc-wbomar review to see if this is the expected behavior.

Reply:
> Let's wait on @tacc-wbomar review to see if this is the expected behavior.

👍   

@tacc-wbomar, I'm really curious how this relates to the use of FileInputDropZoneFormField (which uses FileInputDropZone) in TicketModal/TicketCreateForm

Reply:
I reused that code, should I use FileInputDropZone directly instead?

Reply:
You use `FileInputDropZone` directly in the DataFilesUploadModal and support the larger ("Full Size") view. 👍 

 I just wonder how everything relates to https://jira.tacc.utexas.edu/browse/FP-338, specifically. 

```
3. In the "Small Area" and "Wide But Short" contexts,
When the Listing appears,
Then the Listing must appear overlaid atop the Drop-Zone.

4. In the "Full Size" contexts,
When the Listing appears, — correction made in 2020-06
The Listing must appear vertically below the Drop-Zone.

```
Like this PR adds the "Full Size" context and the the existing `FileInputDropZoneFormField` (used in `TicketModal`/`TicketCreateForm` is the the "Small Area" context??).    but @tacc-wbomar  can shed some light.

Reply:
The current state is not the expected result. It is progress, but it is incomplete and indirect. I have attempted to shed more light. See https://github.com/TACC/Core-Portal/pull/309#issuecomment-802207068.

Reply:
Juan and I chatted. He was meeting the requirements to support multiple layouts, but the properties and component used are not intuitive. I propose we polish and pause the incomplete abstraction.